### PR TITLE
Fix conda errors in release container

### LINKED ISSUE #1744: [BUG]: DFP docker compose fails with conda error
### Version

24.06

### Which installation method(s) does this occur on?

Source

### Describe the bug.

failed to solve: executor failed running [/bin/bash -c source activate morpheus     && mamba env update -n morpheus --file /workspace/conda/environments/examples_cuda-121_arch-x86_64.yaml]: exit code: 1

### Minimum reproducible example

```shell
cd examples/digital_fingerprinting/production
export MORPHEUS_CONTAINER_VERSION="$(git describe --tags --abbrev=0)-runtime"
docker compose build
docker compose up jupyter
```

### Relevant log output

<details><summary>Click here to see error details</summary><pre>

 > [jupyter base 3/5] RUN source activate morpheus     && mamba env update -n morpheus --file /workspace/conda/environments/examples_cuda-121_arch-x86_64.yaml:                          
2.114 Channels:                                                                                                                                                                          
2.114  - conda-forge                                                                                                                                                                     
2.114  - huggingface                                                                                                                                                                     
2.114  - rapidsai                                                                                                                                                                        
2.114  - rapidsai-nightly
2.114  - nvidia
2.114  - nvidia/label/dev
2.114  - pytorch
2.114  - local
2.114 Platform: linux-64
2.114 Collecting package metadata (repodata.json): ...working... failed
6.973 
6.973 UnavailableInvalidChannel: HTTP 404 UNAVAILABLE OR INVALID for channel opt/conda/conda-bld <file:///opt/conda/conda-bld>
6.973 
6.973 The channel is not accessible or is invalid.
6.973 
6.973 You will need to adjust your conda configuration to proceed.
6.973 Use `conda config --show channels` to view your configuration's current state,
6.973 and use `conda config --show-sources` to view config file locations.
6.973 
6.973 As of conda 4.3, a valid channel must contain a `noarch/repodata.json` and
6.973 associated `noarch/repodata.json.bz2` file, even if `noarch/repodata.json` is
6.973 empty. Use `conda index /opt/conda/conda-bld`, or create `noarch/repodata.json`
6.973 and associated `noarch/repodata.json.bz2`.
6.973 
6.973 
------
failed to solve: executor failed running [/bin/bash -c source activate morpheus     && mamba env update -n morpheus --file /workspace/conda/environments/examples_cuda-121_arch-x86_64.yaml]: exit code: 1

</pre></details>

### Full env printout

<details><summary>Click here to see environment details</summary><pre>

[+] Running 1/1
 ! jupyter Warning pull access denied for dfp_morpheus_jupyter, repository does not exist or may require 'docker login': denied: requested access to the resour...                  1.1s 
[+] Building 7.1s (7/10)                                                                                                                                                  docker:default
 => [jupyter internal] load build definition from Dockerfile                                                                                                                        0.1s
 => => transferring dockerfile: 2.38kB                                                                                                                                              0.0s
 => [jupyter internal] load .dockerignore                                                                                                                                           0.1s
 => => transferring context: 213B                                                                                                                                                   0.0s
 => [jupyter internal] load metadata for nvcr.io/nvidia/morpheus/morpheus:v24.06.00rc1-runtime                                                                                      0.0s
 => [jupyter base 1/5] FROM nvcr.io/nvidia/morpheus/morpheus:v24.06.00rc1-runtime                                                                                                   0.0s
 => [jupyter internal] load build context                                                                                                                                           0.1s
 => => transferring context: 12.58kB                                                                                                                                                0.0s
 => CACHED [jupyter base 2/5] WORKDIR /workspace/examples/digital_fingerprinting/                                                                                                   0.0s
 => ERROR [jupyter base 3/5] RUN source activate morpheus     && mamba env update -n morpheus --file /workspace/conda/environments/examples_cuda-121_arch-x86_64.yaml               7.0s
------                                                                                                                                                                                   
 > [jupyter base 3/5] RUN source activate morpheus     && mamba env update -n morpheus --file /workspace/conda/environments/examples_cuda-121_arch-x86_64.yaml:                          
1.770 Channels:                                                                                                                                                                          
1.770  - conda-forge                                                                                                                                                                     
1.770  - huggingface                                                                                                                                                                     
1.770  - rapidsai                                                                                                                                                                        
1.770  - rapidsai-nightly
1.770  - nvidia
1.770  - nvidia/label/dev
1.770  - pytorch
1.770  - local
1.770 Platform: linux-64
1.770 Collecting package metadata (repodata.json): ...working... failed
6.632 
6.632 UnavailableInvalidChannel: HTTP 404 UNAVAILABLE OR INVALID for channel opt/conda/conda-bld <file:///opt/conda/conda-bld>
6.632 
6.632 The channel is not accessible or is invalid.
6.632 
6.632 You will need to adjust your conda configuration to proceed.
6.632 Use `conda config --show channels` to view your configuration's current state,
6.632 and use `conda config --show-sources` to view config file locations.
6.632 
6.632 As of conda 4.3, a valid channel must contain a `noarch/repodata.json` and
6.632 associated `noarch/repodata.json.bz2` file, even if `noarch/repodata.json` is
6.632 empty. Use `conda index /opt/conda/conda-bld`, or create `noarch/repodata.json`
6.632 and associated `noarch/repodata.json.bz2`.
6.632 
6.632 
------
failed to solve: executor failed running [/bin/bash -c source activate morpheus     && mamba env update -n morpheus --file /workspace/conda/environments/examples_cuda-121_arch-x86_64.yaml]: exit code: 1

</pre></details>

### Other/Misc.

_No response_

### Code of Conduct

- [X] I agree to follow Morpheus' Code of Conduct
- [X] I have searched the [open bugs](https://github.com/nv-morpheus/Morpheus/issues?q=is%3Aopen+is%3Aissue+label%3Abug) and have found no duplicates for this bug report

### PR DESCRIPTION
**Pull request by dagardner-nv**
## Description
* Ensures that `/opt/conda/conda-bld` is a valid local conda channel. Without this any additional conda commands would error on an invalid conda channel (#1744). 
* Creates a new intermediate docker stage `runtime_conda_create` allowing it to make use of the read-only `/opt/conda/conda-bld` mount from `conda_bld_morpheus`. Then initializing the `/opt/conda/conda-bld` dir as a valid/empty local channel in the `runtime` stage.
* Update `examples/digital_fingerprinting/production/Dockerfile` to use conda with the libmamba solver.

Closes #1744

## By Submitting this PR I confirm:
- I am familiar with the [Contributing Guidelines](https://github.com/nv-morpheus/Morpheus/blob/main/docs/source/developer_guide/contributing.md).
- When the PR is ready for review, new or existing tests cover these changes.
- When the PR is ready for review, the documentation is up to date with these changes.

### CODE REVIEW

File: docker/Dockerfile [hunk 1]
```diff
@@ -254,10 +245,30 @@ RUN --mount=type=bind,from=conda_bld_morpheus,source=/opt/conda/conda-bld,target
...
+COPY "${MORPHEUS_ROOT_HOST}/conda/environments/*.yaml" "./conda/environments/"
+COPY "${MORPHEUS_ROOT_HOST}/docker" "./docker"
+COPY "${MORPHEUS_ROOT_HOST}/docs" "./docs"
+COPY "${MORPHEUS_ROOT_HOST}/examples" "./examples"
+COPY "${MORPHEUS_ROOT_HOST}/models" "./models"
+COPY "${MORPHEUS_ROOT_HOST}/scripts" "./scripts"
+COPY "${MORPHEUS_ROOT_HOST}/*.md" "./"
+COPY "${MORPHEUS_ROOT_HOST}/LICENSE" "./"
+
+# Ensure the conda-bld directory is indexed even if empty
```
Comment (mdemoret-nv) [inline docker/Dockerfile:268]: Double comment

File: docker/Dockerfile [hunk 2]
```diff
@@ -254,10 +245,30 @@ RUN --mount=type=bind,from=conda_bld_morpheus,source=/opt/conda/conda-bld,target
...
         -c pytorch \
         -c defaults \
-        morpheus &&\
+        morpheus && \
+    /opt/conda/bin/conda env update --solver=libmamba -n morpheus --file \
+        conda/environments/runtime_cuda-${CUDA_MAJOR_VER}${CUDA_MINOR_VER}_arch-x86_64.yaml
 
-    /opt/conda/bin/conda env update --solver=libmamba -n morpheus --file conda/environments/runtime_cuda-${CUDA_MAJOR_VER}${CUDA_MINOR_VER}_arch-x86_64.yaml && \
-    conda clean -afy
+# ============ Stage: runtime ============
```
Comment (mdemoret-nv) [inline docker/Dockerfile:252]: Why is this separated into a separate stage in the build?

**Review by mdemoret-nv — APPROVED** [review]

File: docker/Dockerfile (hunk 2 above)
Comment (dagardner-nv) [reply docker/Dockerfile:252]: Because in the `runtime_conda_create` the `/opt/conda/conda-bld` directory is mounted read-only.

Comment (mdemoret-nv) [reply docker/Dockerfile:252]: I understand now. Resolving

**Pull request merged**